pkg/gadgets/top/ebpf/tracer/tracer.go

```diff
@@ -205,9 +205,27 @@ func (t *Tracer) nextStats() ([]types.Stats, error) {
 		if err != nil {
 			continue
 		}
-		mapSizes[curMapID] = (uint64(mapData.KeySize()) + uint64(mapData.ValueSize())) * uint64(mapData.MaxEntries())
+		defer mapData.Close()
```

Comment:
Review of myself:
`defer mapData.Close()` doesn't do what I think here. It is not bound to block scope, but to function scope.
Not what i want here. Changing...